File: Szulogyerek/szulogyerek.cpp
#include <iostream>
#include <string>


class Szulo
{
	public:
	void message(){
		std::cout << "It's a message\n";
	}
};

class Pulya : public Szulo
{
	public:
	void message2(){
		std::cout << "It's a Pulya message" << "\n";
	}
};


int main(int argc, char const *argv[])
{
	Szulo* szulo = new Szulo();
	Szulo* gyerek= new Pulya();

	szulo->message();
	gyerek->message2();
	delete szulo;
	delete gyerek;
	return 0;
}

File: Szulogyerek/Szulogyerek.java
class Szulo{
	void message(){
		System.out.println("Szülő üzenete.");
	}
}

class Pulya extends Szulo{
	@Override
	void message(){
		System.out.println("Szülő üzenet felülír");
	}
	void message2(){
		System.out.println("Gyerek üzenete");
	}
}

class Szulogyerek{
	public static void main(String[] args) {
		Szulo szulo = new Szulo();
		szulo.message();
		szulo=new Pulya();
		szulo.message();
		//Itt látható, hogy a szülő nem tudja meghívni a gyerek metódusát.Tehát az ősön keresztül, csak az ős üzenetei küldhetők.
		szulo.message2();
		Pulya gyerek=new Pulya();
		gyerek.message();
		gyerek.message2();

	}
}